youtube_dl/extractor/youtube.py

```diff
@@ -3379,110 +3450,35 @@ def _real_extract(self, url):
 
 class YoutubeSearchIE(SearchInfoExtractor, YoutubeBaseInfoExtractor):
     IE_DESC = 'YouTube.com searches'
-    # there doesn't appear to be a real limit, for example if you search for
-    # 'python' you get more than 8.000.000 results
-    _MAX_RESULTS = float('inf')
     IE_NAME = 'youtube:search'
     _SEARCH_KEY = 'ytsearch'
-    _SEARCH_PARAMS = None
+    _SEARCH_PARAMS = 'EgIQAQ%3D%3D'  # Videos only
+    _MAX_RESULTS = float('inf')
```

Comment:
In my opinion, this should be set in `InfoExtractor` as the default. But I don't want to make out of scope changes. So I'm putting this here

Reply:
`SearchInfoExtractor`? What happened with this equivalent change in yt-dlp (extractor/common.py)? 
```
    Instances should define _SEARCH_KEY and optionally _MAX_RESULTS
    """

    _MAX_RESULTS = float('inf')
```
If it didn't break anything we might as well do it. If the CI tests break in new ways then back out.

Reply:
> SearchInfoExtractor? 

Yes, that's what I meant

> What happened with this equivalent change in yt-dlp (extractor/common.py)?
> If it didn't break anything we might as well do it. If the CI tests break in new ways then back out.

No, it won't break anything since each extractor can always override it. It is just a design decision whether you want all search extractors to explicitly define a `_MAX_RESULTS` or if infinity is considered a good default.

Reply:
Let's do it, then, if you like.

Reply:
I don't think it's suitable for this PR

Reply:
Fair enough. It can be done in some pull of useful things from yt-dlp's common.py.